### ISSUE TITLE
URGENT: AWS cloud credits for executable tutorial

### ISSUE THREAD

**Issue by nalenz** [KTH/devops-course#1756]
In order for @elibon99 and me to be able to continue with our executable tutorial from #1678, we need a cloud server where we can install GitLab on-premise.

According to https://docs.gitlab.com/ee/install/requirements.html, GitLab requires at least 4 GB of memory, but the instance type "t2.micro", which is (apart from the even smaller "t1.micro") the only one included in the free trial of AWS, only has 1 GB of RAM. Consequently, we would need to go at least for "t2.medium" with 4 GB RAM, which apparently costs 0.0464 USD per hour.

As we are from KTH, I'm sure that the university can quickly provide cloud credits here, in order to get our project started. As this executable tutorial is highly complex and involves setting up VMs, ssh'ing into them, configuring GitLab etc. in a fully automated fashion, we have to continue as soon as possible, so that we can finish it until the deadline next week.

@monperrus @bbaudry Could you please let us know on how to continue with this, e.g. a voucher code for AWS credits? As both Elias and I work on this (email addresses bonnici@kth.se and nalenz@kth.se), we need these credits for both our AWS accounts.

**Comment by monperrus** [comment]
There should be a free tier plan at Amazon, see https://github.com/KTH/devops-course/issues/25

In principle, you should be able to activate it on your own.

(KTH doesn't have any specific arrangement with AWS).

**Comment by nalenz** [comment]
@monperrus As I mentioned, unfortunately the free tier plan at AWS only includes the "t2.micro" or "t3.micro" instances, which do not provide enough memory (RAM) to be able to run GitLab (see https://aws.amazon.com/free/):

![Bildschirmfoto von 2022-04-12 12-17-54](https://user-images.githubusercontent.com/35401101/162939009-5710de09-89ad-4ef4-9f43-6f98f5d6c25b.png)

In case we would need to pay for AWS by credit card ourselves then, would it be possible to get a refund for these expenses from KTH afterwards, as soon as we have the AWS invoice?

**Comment by monperrus** [comment]
We've never done this, and that's not something usually done in any KTH course.

Are any other free tier packages at Google / Microsoft / Digital Ocean / {others} that you can use for your demo instead?

**Comment by nikolim** [comment]
You can sign up with your KTH-mail for AWS Educate to get some free credit (100$) - hope this helps :)
https://aws.amazon.com/education/awseducate/

**Comment by nalenz** [comment]
Thanks for the hint, Niko! I tried to register for AWS Educate with my KTH email address, but their page seems quite buggy and I didn't receive a verification email, so I contacted their support instead.

In the meantime, we came up with a workaround, which is probably what we'll go for instead: having a local VirtualBox where GitLab is installed and using an AWS virtual machine as an SSH relay station (using SSH remote forwarding/tunneling). Then we can SSH into the local VirtualBox from the remote Google Colab servers for the executable tutorial.

This means that we won't be able to automate the creation of the virtual machines, but at least the installation and setup of GitLab itself.
